oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/StorageID.java

```diff
@@ -0,0 +1,159 @@
+ * friendly interface for various database implementation.
+ */
+@EqualsAndHashCode(of = {
+    "fragments"
+})
+public class StorageID {
+    private final List<Fragment> fragments;
+    /**
+     * Once the storage ID was {@link #build()} or {@link #read()},
+     * this object would switch to the sealed status, no more append is allowed.
+     */
+    private boolean sealed = false;
+
+    public StorageID() {
+        fragments = new ArrayList<>(2);
+    }
+
+    public StorageID append(String name, String value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(name, String.class, false, value));
+        return this;
+    }
+
+    public StorageID append(String name, long value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(name, Long.class, false, value));
+        return this;
+    }
+
+    public StorageID append(String name, int value) {
+        if (StringUtil.isBlank(name)) {
+            throw new IllegalArgumentException("The name of storage ID should not be null or empty.");
+        }
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, name=" + name);
+        }
+        fragments.add(new Fragment(name, Integer.class, false, value));
+        return this;
+    }
+
+    public StorageID appendMutant(String source, long value) {
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, source=" + source);
+        }
+        fragments.add(new Fragment(source, Long.class, true, value));
+        return this;
+    }
+
+    public StorageID appendMutant(final String source, final String value) {
+        if (sealed) {
+            throw new IllegalStateException("The storage ID is sealed. Can't append a new fragment, source=" + source);
+        }
+        fragments.add(new Fragment(source, String.class, true, value));
+        return this;
+    }
+
+    /**
+     * @return the string ID concatenating the values of {@link #fragments} by the underline(_).
+     */
+    public String build() {
+        sealed = true;
+        String[] idParts = new String[fragments.size()];
+        for (int i = 0; i < fragments.size(); i++) {
+            idParts[i] = fragments.get(i).toString();
+        }
+        return StringUtil.join('_', idParts);
```

Comment:
```suggestion
        return com.google.common.base.Joiner.on("_").join(fragments);
```

Reply:
Great, I was searching for this, but failed.